local.go

```diff
@@ -321,8 +337,22 @@ func evaluate(funcs functions, funcInfo *FunctionInfo, evalType EvaluationType,
 			return nil
 		}
 		return ErrMismatch{Type: evalType}
+	case NoEvaluation:
+		if different {
+			// The evaluation point xNext is not equal to location.X, so we
+			// fill location with NaNs to indicate that it does not contain a
+			// valid evaluation result.
+			location.F = math.NaN()
+			for i := range location.X {
+				location.X[i] = math.NaN()
+			}
+			for i := range location.Gradient {
+				location.Gradient[i] = math.NaN()
+			}
+		}
+		return nil
 	default:
-		panic("unreachable")
+		panic(fmt.Sprintf("unknown evaluation type %v", int(evalType)))
```

Comment:
Yeah, I wanted to do this, but it means modifying EvaluationType.String(), because ATM it would panic for unknown evaluation types. I will modify it.